pkg/capabilities/consensus/ocr3/types/aggregator.go

```diff
@@ -24,10 +25,16 @@ type Metadata struct {
 }
 
 // the contract requires exactly 10 bytes for the workflow name
-// the json schema allows for a variable length string <= len(10)
-// pad with trailing spaces to meet the contract requirements
+// the resulting workflow name should be up to 10 bytes long
+// so pad accordingly to meet the contract requirements
 func (m *Metadata) padWorkflowName() {
-	if len(m.WorkflowName) < 10 {
+	b, err := hex.DecodeString(m.WorkflowName)
+	if err == nil && len(b) < 10 {
+		// Each byte is 2 characters, so we need to pad with 0s
+		neededBytes := append(b, make([]byte, 10-len(b))...)
+		m.WorkflowName = hex.EncodeToString(neededBytes)
+	} else if len(m.WorkflowName) < 10 {
+		// Pad with spaces
 		suffix := strings.Repeat(" ", 10-len(m.WorkflowName))
```

Comment:
This should never be the case. You also don't really have to hex-decode. Just pad with "0" chars up to length 20.

Reply:
cc @krehermann  - cap_encoder.go in core validates that and fails it it's not a valid hex string that decodes to 10 bytes. We should require it here too.

Reply:
ok. if it's got to be that then let's move the validation to this struct. can be a follow up change.

code comments aren't sufficient if there are restrictions on the values. and doesn't make sense to have validation far away in a different repo. @vyzaldysanchez can you do that as a follow up (write a Validate method for this struct)

Reply:
sure thing!